rt4914/KlinicCon-Backend
# Fixes #19: Creation of patient & doctor profile with associations

1.Creation of patient profile migration file
2.Creation of doctor profile migration file
3.Creation of new migration for adding references to patient and doctor profile
4.Two models patient_profile.rb and doctor_profile.rb created to have associations mentioned with respective columns 5.user model added with belongs_to association for patient and doctor profile 
6.Models institute.rb, specialization.rb, address.rb is updated with has_one association with doctor_profile

Resolves #19 

### Description
Creation of patient & doctor profile models with the right associations present for other models like institute, specialization  & address, user model is also updated with associations related to patient and doctor models


### Type of change



### How to test this PR?

After running migrations, and using rails c, run below queries to check the columns for the respective tables

irb(main):001> PatientProfile.column_names
=>
["id",
 "name",
 "gender",
 "date_of_birth",
 "blood_group",
 "address_id",
 "created_at",
 "updated_at"]
irb(main):002>

irb(main):002> DoctorProfile.column_names
=>
["id",
 "name",
 "gender",
 "date_of_birth",
 "specialization_id",
 "degree_id",
 "institute_id",
 "year_of_completion",
 "year_of_experience",
 "created_at",
 "updated_at"]

irb(main):003> User.column_names
=>
["id",
 "email",
 "encrypted_password",
 "reset_password_token",
 "reset_password_sent_at",
 "remember_created_at",
 "created_at",
 "updated_at",
 "confirmation_token",
 "confirmed_at",
 "confirmation_sent_at",
 "unconfirmed_email",
 "jti",
 "patient_profile_id",
 "doctor_profile_id"]
